### PR TITLE
change vjp to paddle.grad

### PR DESCRIPTION
**Pull request by betterpig**
### PR types

Others
### PR changes

Others
### Describe

Others

### CODE REVIEW

File: python/paddle/incubate/optimizer/functional/utils.py [hunk 1]
```diff
@@ -87,8 +87,9 @@ def _value_and_gradient(f, x, v=None):
         gradient: a tensor that holds the function gradients.  
     """
     if paddle.in_dynamic_mode():
-        value, gradient = vjp(f, x, v=v)
-        gradient = gradient[0]
+        x.stop_gradient = False
```
Comment (levi131) [inline python/paddle/incubate/optimizer/functional/utils.py]: 这里x原始会有stop_gradients=True的情况么，这里会有副作用

Comment (levi131) [reply python/paddle/incubate/optimizer/functional/utils.py]: if x.stop_gradient:
    x = assign(x)
else:
    x= x.detach()
    x.stop_gradient=False

Comment (cxxly) [reply python/paddle/incubate/optimizer/functional/utils.py]: > if x.stop_gradient: x = assign(x) else: x= x.detach() x.stop_gradient=False

应该是 if not x.stop_gradient吧

```python
    if not x.stop_gradient:
        x = paddle.clone(x) # assign也行
    else:  # use detach to share memory when no need gradients.
        x = x.detach()
        x.stop_gradient = False
     
```

Comment (levi131) [reply python/paddle/incubate/optimizer/functional/utils.py]: 是的，我写错了

**Review by levi131 — APPROVED** [review]
LGTM
